packages/metatransaction-broadcaster/MetatransactionBroadcaster.js

```diff
@@ -244,13 +230,172 @@ class MetatransactionBroadcaster {
     try {
       const tx = possibleVotingRep.interface.parseTransaction({ data: txData });
       if (tx.signature === "finalizeMotion(uint256)") {
-        allowed = false;
+        // eslint-disable-next-line no-underscore-dangle
+        const motion = await possibleVotingRep.getMotion(tx.args._motionId);
+        // Get the motion
+        let motionTarget = motion.altTarget;
+        if (motionTarget === "0x0000000000000000000000000000000000000000") {
+          motionTarget = await possibleVotingRep.getColony();
+        }
+        // Is the motion doing something we'd allow? Duplicated logic from main function
+
+        const addressValid = await this.isAddressValid(motionTarget);
+        // It's possible it's not a valid address, but could be a valid transaction with a token.
+        const validTokenTransaction = await this.isTokenTransactionValid(motionTarget, motion.action, userAddress);
```

Comment:
This looks similar to other code snippets elsewhere. Perhaps a function which takes the arguments, performs all these checks, and returns a single boolean?

Reply:
The problem is that the other place that this logic is used, it uses the details of what specific checks failed to change the message returned, so returning a single boolean isn't adequate.

Reply:
Hmm. Could we return `(result, err)` then? Are the error messages consistent (or could they be made consistent)? Alternatively we could just return `err === undefined` if the checks pass.